packages/flutter_test/lib/src/tree_traversal.dart

```diff
@@ -0,0 +1,167 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/semantics.dart';
+import 'package:flutter/widgets.dart';
+
+/// Provides an iterable that efficiently returns all the elements
+/// rooted at the given element. See [CachingIterable] for details.
+///
+/// This method must be called again if the tree changes. You cannot
+/// call this function once, then reuse the iterable after having
+/// changed the state of the tree, because the iterable returned by
+/// this function caches the results and only walks the tree once.
+///
+/// The same applies to any iterable obtained indirectly through this
+/// one, for example the results of calling `where` on this iterable
+/// are also cached.
+Iterable<Element> collectAllElementsFrom(
+  Element rootElement, {
+  // TODO(pdblasi-google): Deprecate `skipOffstage`. @Deprecated(
+  //   'Use includeOffstage instead. '
+  //   'This allows us to be consistent with the new FinderBase API. '
+  //   'This feature was deprecated after v3.11.0-0.0.pre.'
+  // )
+  bool? skipOffstage,
+  bool includeOffstage = false,
+}) {
+  if (skipOffstage != null) {
+    includeOffstage = !skipOffstage;
+  }
+  return CachingIterable<Element>(_DepthFirstElementTreeIterator(rootElement, includeOffstage));
+}
+
+/// Provides an iterable that efficiently returns all the [SemanticsNode]s
+/// rooted at the given [SemanticsNode]. See [CachingIterable] for details.
+///
+/// This method must be called again if the semantics change. You cannot call
+/// this function once, then reuse the iterable after having changed the state
```

Comment:
first sentence: ~method~ -> `function`

Reply:
Good catch on the inconsistency! I'm going to go the other route though and update the "function"s to "method"s. "Method" seems to be more common in the rest of the docs.

Reply:
Hmm, in my book a _method_ is a function that is inside a class.

See also [this SO answer](https://stackoverflow.com/questions/155609/whats-the-difference-between-a-method-and-a-function). This seems to be quite generally agreed upon.

But I'm not gonna die on this hill haha

Reply:
Fair point, and it's consistent with the Dart docs as well. The docs I checked against didn't have any top level methods, but where there _are_ more top level methods it looks like `function` is at least common. Now I'm curious on how consistent we are with that...

Either way, I'll fix it here while we're in the area and you brought it up.

Reply:
Done!